CacheManager.StackExchange.Redis.SelfCleaning/SelfCleaningRedisCacheHandle.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using CacheManager.Core;
+using CacheManager.Core.Internal;
+using CacheManager.Core.Logging;
+using CacheManager.Redis;
+using CacheManager.StackExchange.Redis.SelfCleaning.Core;
+using CacheManager.StackExchange.Redis.SelfCleaning.Timers;
+using StackExchange.Redis;
+
+namespace CacheManager.StackExchange.Redis.SelfCleaning
+{
+    public class SelfCleaningRedisCacheHandle<TCacheValue> : RedisCacheHandle<TCacheValue>, IStartable
+    {
+        private readonly IDatabase _redisDatabase;
+        private readonly ITimer _cleanupTimer;
+        private readonly TimeSpan _slidingExpiration;
+        private readonly int _databaseId;
+        
+        public SelfCleaningRedisCacheHandle(ICacheManagerConfiguration managerConfiguration,
+            CacheHandleConfiguration configuration, ILoggerFactory loggerFactory, ICacheSerializer serializer) : base(
+            managerConfiguration, configuration, loggerFactory, serializer)
+        {
+            RedisConfiguration redisConfiguration = RedisConfigurations.GetConfiguration(configuration.Key);
+
+            if (!(redisConfiguration is SelfCleaningRedisConfiguration selfCleaningRedisConfiguration))
+            {
+                throw new ArgumentException("The given configuration is not of type SelfCleaningRedisConfiguration",
+                    nameof(configuration));
```

Comment:
Shouldnt it be:

![image](https://user-images.githubusercontent.com/62211836/76997811-940cdc80-695c-11ea-8bfa-44ce97034f9c.png)


Beacuse that the variable that was of the wrong type